sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpTransaction.java

```diff
@@ -42,6 +44,11 @@ public ByteBuffer getTransactionId() {
      * @return string representation of the transaction id.
      */
     public String toString() {
-        return new String(transactionId.array(), StandardCharsets.UTF_8);
+        if (transactionId.hasArray()) {
+            return new String(transactionId.array(), transactionId.arrayOffset() + transactionId.position(),
+                transactionId.remaining(), StandardCharsets.UTF_8);
```

Comment:
Stupid question, why start from `byteBuffer.position()`?

Is it common that someone sets the position? And if someone did, do we intent to read the remaining for the whole?

Reply:
Not a stupid question at all 😃 

While it is unknown how often an `arrayOffset` and/or `position` to be set in a passed `ByteBuffer` using them ensures we're only accessing what is still considered readable by the `ByteBuffer`. 

For example, I create a `ByteBuffer` with a 4096 byte array, and I know the array has special metadata in the first 128 bytes. So, I'll create a wrapping `ByteBuffer` offset by 128 bytes, so its effective reading range are the bytes [128, 4096). At this point the `toString` logic highlighted will be based on that range, `new String(byte[], 128, 4096-128, UTF_8)` (pardon any off-by-one errors as I'm free-handing this). But, let's say next that I know the range [128, 1024) is either junk data or something unrelated to the string, I'll make the new position 1024-128 as there is an offset of 128. Now, the `toString` becomes `new String(byte[], 128+896, 4096, UTF_8)`.

For the second question, this is more difficult on whether the `ByteBuffer`'s position should be mutated, in this case I'd say no as `toString` is more a representation and may result in broken runtime behavior if `transactionId` is fully consumed by it. Also, when debugging, many IDEs will call `toString` when inspecting the object in the debug window, likely breaking debugging.